lib/route_consistency_checker.rb

```diff
@@ -0,0 +1,143 @@
+require "set"
+require "gds_api/router"
+
+class RouteConsistencyChecker
+  attr_reader :errors
+
+  def initialize(routes, router_data)
+    @routes = routes
+    @router_data = router_data
+    @checked_routes = Set.new
+    @errors = Hash.new { |hash, key| hash[key] = [] }
+  end
+
+  def check_content
+    content_items_to_check.each do |content_item|
+      content_item.redirects.each do |redirect|
+        check_redirect(content_item, redirect)
+      end
+
+      content_item.routes.each do |route|
+        check_route(content_item, route)
+      end
+    end
+  end
+
+  def check_routes
+    unchecked_routes.each do |path|
+      next if router_data.include?(path)
+
+      route = routes.fetch(path)
+      next if route.disabled
+      next if route.handler == "gone"
+      next if route.handler == "redirect" && !route.backend_id.present?
+
+      content_item = find_content_item(route.incoming_path)
+      next if content_item
+
+      errors[route.incoming_path] << "No content item available."
+    end
+  end
+
+private
+
+  attr_reader :routes, :router_data, :checked_routes
+  attr_writer :errors
+
+  def content_items_to_check
+    ContentItem
+      .where(:content_id.nin => ["", nil])
+      .where(:schema_name.not => /^placeholder/)
+  end
+
+  def unchecked_routes
+    Set.new(routes.keys) - checked_routes
+  end
+
+  def get_route(path)
+    begin
+      route = routes.fetch(path.to_sym)
+      checked_routes.add?(path.to_sym)
+      route
+    rescue KeyError
+      errors[path] << "Path (#{path}) was not found!"
+      nil
+    end
+  end
+
+  def find_content_item_for_field(path, kind)
+    items = ContentItem.where("#{kind}s.path" => path).entries
+    return nil if items.empty?
+    return items.first if items.length == 1
+    errors[path] << "Multiple content items returned for #{kind}."
+    nil
```

Comment:
I feel like this would read better in a if else block - to me it feels like it's supposed to return nil whereas you want it to look more like multiple items is an exceptional circumstance